amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -660,7 +657,30 @@ public static AmpActivityVersion initializeActivity(AmpActivityVersion act) thro
    return ActivityVersionUtil.groupOutputCollection(outputCollection);
 }
  
-    
+    public static Map<Long, Map<String, List<CompareOutput>>> compareActivities(List<Long> activityId) throws Exception {
+        Map<Long,Map<String,List<CompareOutput>>>  listOfActivities = new HashMap<Long,Map<String,List<CompareOutput>>>();
+         ArrayList<Long> activityList =  new ArrayList(activityId);
+         Iterator <Long>iter = activityList.iterator();
+         while(iter.hasNext())
+         {
+             Long  currentActivity = iter.next();
+             Session session = PersistenceManager.getCurrentSession();
+             
+             AmpActivityVersion ampActivityOne = (AmpActivityVersion) session.load(AmpActivityVersion.class, currentActivity);
+
+             AmpActivityVersion ampActivityTwo = ActivityUtil.getPreviousVersion(ampActivityOne);
+           
+             
+             if(ampActivityTwo==null) 
```

Comment:
good check, to avoid a null pointer. Now that you will reuse the code you had in the previous ticket, lets add this check in that ticket so we can handle when there is nothing to compare to